SetUp/autoDefine/autoDefine.py

```diff
@@ -314,16 +314,30 @@ def inputBuilder( directories, options, keep_going ):
         defInp.close()
 
 
-        #Now copy the define input file over and run define with it. 
+        #Now move the define input file over and run define with it. 
         os.rename('def.input',dirs+'/def.input')
+        copyfile('templateCosmo.input',dirs+'/templateCosmo.input')
         os.chdir(dirs)
         if os.path.exists('control'):
             os.remove('control')
 
         p=sp.Popen('define < def.input > def.out',shell=True)
         p.wait()
+        q=sp.Popen('cosmoprep < templateCosmo.input > cosmoprep.out',shell=True)
+        q.wait()
+
+        #Manually add $disp3 line at end of control file
+        p=sp.Popen("sed -i 's/\$end/\$disp3 bj\\n$end/' control",shell=True)
```

Comment:
Do you need those backslashes on the $? Think single quotes take care of that and sed doesn't need. Will check